macrosynergy/download/dataquery.py

```diff
@@ -285,540 +317,629 @@ def _get_token(self) -> str:
-                        return None
-                else:
-                    logger.warning(
-                        f"respone returned with HTTP Status Code {int(msg['status_code'])}. "
-                        f"response : {response}, "
-                        f"status_code : {int(msg['status_code'])}, "
-                        f"msg : {msg}, "
-                        f"url : {url}, "
-                        f"params : {params},"
-                        f"dq_api.Interface.last_url : {self.last_url}, "
-                        f"status_code : {int(msg['status_code'])} "
-                    )
-                    raise InvalidResponseError(
-                        f"Invalid response from DataQuery. response : {response}"
-                        f"status_code : {int(msg['status_code'])}, "
-                        f"msg : {msg}, "
-                        f"url : {url}, "
-                        f"params : {params},"
-                    )
+        Not yet implemented.
+        """
+        raise NotImplementedError("This method has not been implemented yet.")
 
-            except ConnectionResetError:
-                conxn_errors += 1
-                time.sleep(0.05)
-                logger.warning(
-                    f"Server error: will retry. Retry number: {conxn_errors+invalid_responses}. "
-                    f"ConnectionResetError count: {conxn_errors}, "
-                    f"invalid_responses count: {invalid_responses}, "
-                    f"dq_api.Interface.last_url : {self.last_url}, "
-                    f"dq_api.Interface.last_response : {self.last_response}, "
-                )
-                continue
-            except ValueError:
-                invalid_responses += 1
-                time.sleep(0.05)
-                logger.warning(
-                    f"Server error: Invalid response received. Retry number: {conxn_errors+invalid_responses}. "
-                    f"ConnectionResetError count: {conxn_errors}. "
-                    f"invalid_responses count: {invalid_responses}. "
-                    f"response : {response}, "
-                    f"status : {status}, "
-                    f"msg : {msg}, "
-                    f"url : {url}, "
-                    f"params : {params}, "
-                    f"dq_api.Interface.last_url : {self.last_url}"
-                )
-            else:
-                logger.info(f"Request successful. {track_id}")
-                if select in response.keys():
-                    results.extend(response[select])
-
-                if "links" not in response.keys():
-                    raise InvalidResponseError(
-                        f"Invalid response from DataQuery. response : {response}"
-                        f"links missing from response.keys():"
-                        f"Status Code: {int(msg['status_code'])}"
-                        f"msg : {msg}"
-                        f"url : {url}"
-                    )
+    def validate_download_args(
```

Comment:
Didn't we have this validation in jpmaqs.py as well? Should only be present in one place (and if necessary re-used).

Reply:
Similar, but not exactly the same (different args). The date checks are present in both, as even for a DQ interface it makes sense to accept input in ISO8601. 
I could write a validation check to be shared accross files, but it would only complicate the function beyond utility.